src/routes/add-article.ts

```diff
@@ -1,27 +1,30 @@
 import createHttpError from 'http-errors';
 import { constants } from 'http2';
-import jsonld from 'jsonld';
-import { JsonLdArray } from 'jsonld/jsonld-spec';
 import { Next } from 'koa';
-import { schema } from 'rdf-namespaces';
-import uniqueString from 'unique-string';
 import { AppContext, AppMiddleware } from '../app';
 import NotAnArticle from '../errors/not-an-article';
+import { rdf, schema } from '../namespaces';
 
 export default (): AppMiddleware => (
   async ({ articles, request, response }: AppContext, next: Next): Promise<void> => {
-    const [article] = await jsonld.expand(request.body) as JsonLdArray;
-    if ('@id' in article) {
-      throw new createHttpError.Forbidden(`Article IDs must not be set ('${article['@id']}' was given)`);
+    const foundArticles = request.dataset.match(null, rdf.type, schema.Article);
+
+    if (foundArticles.size > 1) {
+      throw new createHttpError.BadRequest('Multiple articles found');
     }
-    if (!(schema.name in article) || article[schema.name].length === 0) {
-      throw new createHttpError.BadRequest(`Article must have at least one ${schema.name}`);
+
+    if (foundArticles.size === 0) {
+      throw new createHttpError.BadRequest(`No ${schema.Article.value} found`);
     }
 
-    article['@id'] = `_:${uniqueString()}`;
+    const id = [...foundArticles][0].subject;
```

Comment:
This ID needs to be validated that it's a blank node, then replace with a new unique ID.